### PR TITLE
Show warning about prerendered pages

### LINKED ISSUE #51: Show warning in pre-rendered page.
If seed.org is open in a browser with JavaScript disabled or without WebAssembly support, then searching and switching the color scheme does not work quietly. We should display a warning that the browser does not support WebAssembly.

### PR DESCRIPTION
**Pull request by akhilman**
Show a warning, disable the search, and disable the dark/light
mode switch for pre-processed pages, so as not to confuse users
with disabled scripts or browsers without support for web assembly.

Resolves #51

### CODE REVIEW

File: crate/src/page/partial/content_control_panel.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ use crate::{
     generated::css_classes::C, page::partial::image, Guide, Mode, Model, Msg,
     Urls,
 };
-use seed::{a, attrs, div, empty, prelude::*, span, style, C};
+use seed::{a, attrs, div, empty, prelude::*, span, style, C, IF};
```
Comment (MartinKavik) [inline crate/src/page/partial/content_control_panel.rs:8]: I think we can write standard `use seed::{prelude::*, *}` and force Clippy to ignore wildcard imports.

File: crate/src/page/partial/content_control_panel.rs [hunk 2]
```diff
@@ -73,11 +73,11 @@ fn view_mode_toggle(in_prerendering: bool, mode: Mode) -> Node<Msg> {
...
                 C.border_gray_400,
-                C.cursor_pointer,
                 C.rounded_full,
-                C.hover__underline,
-                C.hover__text_gray_700,
-                C.hover__border_gray_600,
+                IF!(!in_prerendering => C.cursor_pointer),
+                IF!(!in_prerendering => C.hover__underline),
+                IF!(!in_prerendering => C.hover__text_gray_700),
+                IF!(!in_prerendering => C.hover__border_gray_600),
```
Comment (MartinKavik) [inline crate/src/page/partial/content_control_panel.rs:80]: Does it make sense to write it like `IF!(not(in_prerendering) => vec![C.xxx, C.yyy])`?

Comment (akhilman) [reply crate/src/page/partial/content_control_panel.rs:80]: I forgot about this option. I'll fix this.

File: crate/src/page/partial/guide_list.rs [hunk 3]
```diff
@@ -164,11 +164,13 @@ fn view_search(model: &Model) -> Node<Msg> {
                     C.px_2,
                     C.pl_8,
                     C.appearance_none,
+                    IF!(model.in_prerendering => C.opacity_50),
```
Comment (akhilman) [inline crate/src/page/partial/guide_list.rs:167]: Is it OK tu use opacity for disabled search? Should I replace colors instead?

**Comment by MartinKavik** [comment]
I shouldn't merge it without proper testing. There is now ugly flashing blue window on seed-rs.org, please fix it.

File: crate/src/page/partial/content_control_panel.rs [hunk 4]
```diff
@@ -73,11 +73,14 @@ fn view_mode_toggle(in_prerendering: bool, mode: Mode) -> Node<Msg> {
                 C.text_gray_500,
                 C.border,
                 C.border_gray_400,
-                C.cursor_pointer,
                 C.rounded_full,
-                C.hover__underline,
-                C.hover__text_gray_700,
-                C.hover__border_gray_600,
+                IF!(!in_prerendering => vec![
```
Comment (MartinKavik) [inline crate/src/page/partial/content_control_panel.rs:77]: `!` => `not` to make it consistent with `not(model.pre..` in `guide_list.rs`

File: crate/src/page/partial/guide_list.rs [hunk 5]
```diff
@@ -155,7 +155,12 @@ fn view_search(model: &Model) -> Node<Msg> {
                     C.bg_green_100,
                     C.text_sm,
                     C.text_green_800,
-                    C.placeholder_green_800,
+                    IF!(not(model.in_prerendering) => vec![
+                        C.placeholder_green_800,
+                    ]),
+                    IF!(model.in_prerendering => vec![
+                        C.placeholder_green_400,
+                    ]),
```
Comment (MartinKavik) [inline crate/src/page/partial/guide_list.rs:163]: We don't need `vec`s here.

**Comment by MartinKavik** [comment]
And I've added some comments, because I wanted to fixed them by myself after merging but now I don't want to change the code under your hands.

**Pull request merged**